interface/main/mobile/m_cal.php

```diff
@@ -0,0 +1,258 @@
+$display = "cal";
+
+/*/remember state long after session
+$uspfx              = substr(__FILE__, strlen($webserver_root)) . '.';
+$setting_date       = prevSetting($uspfx, 'setting_date', 'setting_date', $day);
+$setting_month      = prevSetting($uspfx, 'setting_month', 'setting_month', $month);
+$setting_year       = prevSetting($uspfx, 'setting_year', 'setting_year', $year);
+*/
+?><!doctype html>
+<html style="cursor: pointer;">
+<?php
+    common_head();
+?>
+
+<body style="background-color: #fff;" >
+<?php common_header($display); ?>
+
+    <?php
+    if (!empty($_GET['eid'])) {
+        echo "<script>$(document).ready(function () { ScrollIt(); }); </script>";
+    }
+        
+        $day        = $_GET['day'];
+        $month      = $_GET['month'];
+        $provider   = $_GET['provider'];
+        $eid        = $_GET['eid'];
+
+        //if no provider is assigned, return all appointments
+    if (is_null($provider) && empty($provider) || !is_numeric($provider)) {
+        $provider = '';
+    }
+        
+        //if no day and month are provided, use the current date
+    if ((is_null($day) || empty($day) && is_null($month) || empty($month)) || !is_numeric($day) || !is_numeric($month)) {
+        $thismonth  = ( int ) date("m");
+        $thisyear   = date("Y");
+        $today      = date("j");
+    } else {
+        $thismonth = $month;
+        $thisyear = date("Y");
+        $today = $day;
+    }
+        // find out the number of days in the month
+        $numdaysinmonth = cal_days_in_month(CAL_GREGORIAN, $thismonth, $thisyear);
+        
+        // create a calendar object
+        $jd = cal_to_jd(CAL_GREGORIAN, $thismonth, date(1), $thisyear);
+        
+        // get the start day as an int (0 = Sunday, 1 = Monday, etc)
+        $startday = jddayofweek($jd, 0);
+        
+        // get the month as a name
+        $monthname = jdmonthname($jd, 1);
+    ?>
+    <div id="gb-main" class="container-fluid">
+        <form id="save_media" name="save_media" action="#" method="post" enctype="multipart/form-data">
+
+            <div class="row">
+                <div class="col-xs-12 col-sm-12 col-md-12 col-lg-12 text-center">
+                    <img src="<?php echo $webroot; ?>/public/images/calendar.png" id="head_img" alt="OpenEMR <?php echo xla('Calendar'); ?>">
+                </div>
+                <div class="col-xs-12 col-sm-4 col-md-4 col-lg-4 text-center">
+                    <div class="row text-center">
+                        <div class="col-xs-12 col-sm-12 col-md-12 col-lg-12 custom-file-upload">
+                            <table WIDTH="100%">
+                                <tr>
+                                    <td colspan="2" bgcolor="#C0C0C0"><div align="center"><a href=m_cal.php?provider=<?php echo $provider ?>&day=<?php echo $today ?>&month=<?php echo $thismonth-1 ?>><?php echo jdmonthname(cal_to_jd(CAL_GREGORIAN, $thismonth-1, date(1), $thisyear), 0) ?></a></div></td>
```

Comment:
```
<td colspan="2" bgcolor="#C0C0C0"><div align="center"><a href=m_cal.php?provider=<?php echo attr_url($provider) ?>&day=<?php echo attr_url($today) ?>&month=<?php echo attr_url($thismonth-1) ?>><?php echo text(jdmonthname(cal_to_jd(CAL_GREGORIAN, $thismonth-1, date(1), $thisyear), 0)) ?></a></div></td>
```
this is a function that basically does attr(urlencode()), which will urlencode get parameters when you manually build the path like above and then also html escape it.